packages/ui/src/Select.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import * as React from 'react';
+
+/**
+ * The class name added to all UI components.
+ */
+const UI_CLASS = 'jp-ui';
+
+/**
+ * The class name added to wrapper div.
+ */
+const WRAPPER_CLASS = 'jp-select-wrapper';
+
+/**
+ * The class name added to styled select.
+ */
+const STYLED_CLASS = 'jp-mod-styled';
+
+/**
+ * The class name added to a focused select element.
+ */
+const SELECT_FOCUSED_CLASS = 'jp-mod-focused';
+
+/**
+ * The properties of the Select component.
+ */
+export interface ISelectProps {
+  children?: any;
+  className?: string;
+  onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  onKeydown?: (event: React.KeyboardEvent<HTMLSelectElement>) => void;
+  options: string[];
+  selected?: string;
+  [prop: string]: any;
```

Comment:
Same Q with these props, can we have this typed somehow?

Reply:
Maybe we could type them by adding some prop called like `selectProps` which is a partial of the select element props? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/232a917561aa9ec0098a626f8a668a52c7e73469/types/react/index.d.ts#L2338

If you think the user should be able to override them